# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1426437279 14400
#      Sun Mar 15 12:34:39 2015 -0400
# Node ID 4910b70d73284dd4388aec081ee4387b6257d57d
# Parent  f7a805f02723cf06e4dc4c6d2bd81645680bfbd7
Ensure correct coordinates are available in windowbuttondownfcn callback

* __init_fltk__.cc (plot_window::handle): Ensure the "windowbuttondownfcn"
callback is called after the axes property "currentpoint" is updated.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -1461,26 +1461,31 @@ private:
                         && Fl::event_ctrl ())
                        || Fl::event_button () == FL_RIGHT_MOUSE)
                 fp.set_selectiontype ("alternate");
               else if (Fl::event_clicks ())
                 fp.set_selectiontype ("open");
               else
                 fp.set_selectiontype ("normal");
 
-              if (fp.get_windowbuttondownfcn ().is_defined ())
-                fp.execute_windowbuttondownfcn (Fl::event_button ());
-
               gh = pixel2axes_or_ca (pos_x, pos_y);
 
               if (gh.ok ())
                 {
                   ax_obj = gh_manager::get_object (gh);
                   set_axes_currentpoint (ax_obj, pos_x, pos_y);
+                }
 
+              // Ensure windowbuttondownfcn is called after currentpoint
+              // is updated but before calling buttondownfcn.
+              if (fp.get_windowbuttondownfcn ().is_defined ())
+                fp.execute_windowbuttondownfcn (Fl::event_button ());
+
+              if (gh.ok ())
+                {
                   int ndim = calc_dimensions (ax_obj);
 
                   if (ndim == 3)
                     rotate->activate ();
                   else // ndim == 2
                     rotate->deactivate ();
 
                   fp.set_currentobject (ax_obj.get_handle ().value ());
